platform/linux-generic/arch/aarch64/cpu_flags.c

```diff
@@ -8,11 +8,738 @@
 
 #include <string.h>
 #include <stdlib.h>
+#include <sys/auxv.h>
+#include <asm/hwcap.h>
 
 #include <odp/api/hints.h>
 #include <odp_debug_internal.h>
 #include "cpu_flags.h"
 
+typedef struct {
+	const char *feat_flag;
+	const unsigned int hwcap_field;
+	bool valid;
+} hwcap_feat_flag_t;
+
+static hwcap_feat_flag_t hwcap_flags[] = {
+#ifdef HWCAP_FP
+	{
+		/* Floating-point Extensions */
+		.feat_flag = "FEAT_FP",
+		.hwcap_field = HWCAP_FP,
+		.valid = 1,
+	},
+#else
+	{
+		.feat_flag = "FEAT_FP",
+		.valid = 0,
+	},
+#endif
```

Comment:
These ifdefs could be simplified a bit as members without initializers are automatically zeroed. For example:
```
	{
		/* Floating-point Extensions */
		.feat_flag = "FEAT_FP",
#ifdef HWCAP_FP
		.hwcap_field = HWCAP_FP,
		.valid = 1,
#endif
	},
```

Reply:
Looks good. I will make the change. Thanks Matias!